api/Controllers/HotelsController.cs

```diff
@@ -73,9 +73,17 @@ public async Task<IActionResult> GetHotel([FromRoute] int hotelId)
         }
 
         [HttpGet]
-        public async Task<IActionResult> GetHotelCards([FromQuery] PageParameters pageParameters)
+        public async Task<IActionResult> GetHotelCards(
+            [FromQuery] PageParameters pageParameters, 
+            [FromQuery] FilterParameters filterParams
+        )
         {
-            HotelCard[] hotelCards = await _hotelsDb.Hotels
+            var filteredHotelCards = _hotelsDb.Hotels.AsQueryable();
+            if (!string.IsNullOrEmpty(filterParams.Name))
+            {
+                filteredHotelCards = filteredHotelCards.Where(h => h.Name.ToLower().Contains(filterParams.Name.ToLower()));
```

Comment:
Do you really need both ToLower calls here? If still yes - what are the alternatives?

Reply:
There was a comment here like 'Discuss with Andrew'. We tried approach with StringComparer with Alex, it did not work. Would like to discuss it with you also